omaha/mi_exe_stub/mi.cc

```diff
@@ -718,6 +756,11 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE, LPSTR lpCmdLine, int) {
 
   omaha::StorePathToRegForPromoCode(lpCmdLine);
 
+  const CString app_guid = omaha::ReadAppGuidFromTag(hInstance);
+  omaha::StoreAppGuidToReg(app_guid);
```

Comment:
This could have race condition if user launches different channel stub installer simultaneously.

Reply:
Is running two stub installers simultaneously a scenario that we support? It seems like it would cause us other problems, e.g. with the promo code that we store in the registry. Even if I put a mutex here to ensure that the app_guid that I read is the same app_guid that I write, it seems that there is still an opportunity to confuse our code. For example: installer A reads an app_guid of "foo" out of the tag and writes "foo" to the registry, then installer B runs and writes "bar" to the registry, then installer A accidentally uses "bar" instead of "foo". Not sure how we would avoid that?

Reply:
You're right. Passing info via registry could have same issue.
Oh.. I found that `App class` has `app_guid_string()` method.
If that method returns same value as in registry, we can use that method.
But not sure this approach doesn't have same issue.
Actually, I didn't test run installer simultaneously.

Reply:
I think this case would be very rare.
So, how about handle this by separate issue?